massgov/mayflower
# DP-10086: SVG Icon Proof of Concept.

Any PRs being created needs a changelog.txt file before being merged into dev. See: [Change Log Instructions](https://github.com/massgov/mayflower/blob/dev/docs/change-log-instructions.md)


## Description
A new component has been created, Icon, that replaces all the previously used SVG components. This allows us to store all SVG files in one directory within the component directory and pass props to modify that SVG's rendering. It makes use of [svg-sprite-loader](https://github.com/kisenka/svg-sprite-loader) to generate a SVG sprite on the fly that contains only the SVG assets that are needed on the page at the time.

All previously existing SVG components have been removed and replaced with Icon, so testing will need to cover quite a lot of components here.

svg-sprite-loader strips off the original attributes from the original svg element. This means that in some cases, you'll need to pass the svg's width and height to its svgWidth and svgHeight props. We could also make a new prop for CSS classes in the future (or now if needed).

Once this PR is reviewed and accepted, a new PR on massgov search will be needed to handle the changing of components.

## Related Issue / Ticket

- [JIRA issue](https://jira.mass.gov/browse/DP-10086)

## Steps to Test


1. Pull the branch and run `npm install`.
2. Start up storybook (`npm run start`) and verify that there is now an Icon component under atoms/icons.
3. Click to the Icon component. Using knobs, verify that each SVG icon renders properly.
4. For each SVG icon, find where it exists in other components and verify that everything looks the same as before.

## Screenshots

![screen shot 2018-08-27 at 11 09 04 am](https://user-images.githubusercontent.com/5257660/44673557-e3499000-a9f9-11e8-842d-95337e5725ba.png)

<img width="633" alt="screen shot 2018-08-27 at 11 09 26 am" src="https://user-images.githubusercontent.com/5257660/44673562-e775ad80-a9f9-11e8-9e68-0d2507abe087.png">

![screen shot 2018-08-27 at 1 06 03 pm](https://user-images.githubusercontent.com/5257660/44673589-feb49b00-a9f9-11e8-85ef-4639aff99d58.png)
